core/include/detray/builders/bin_fillers.hpp

```diff
@@ -122,10 +122,13 @@ struct bin_associator {
                                 const transform_container &transforms,
                                 const mask_container &masks,
                                 const context_t ctx, Args &&...) const -> void {
+
+        using scalar_t = dscalar<typename grid_t::algebra_type>;
+
         // Fill the surfaces into the grid by matching their contour onto the
         // grid bins
-        bin_association(ctx, surfaces, transforms, masks, grid, {0.1f, 0.1f},
-                        false);
+        bin_association(ctx, surfaces, transforms, masks, grid,
+                        std::array<scalar_t, 2>{0.1f, 0.1f}, false);
```

Comment:
Was this change necessary? What sort of error/warning did you get without it? 🤔

Reply:
The scalar type here is deduced now instead of using the global one. That only works, if I spell out the array type.